src/GoodsList.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class GoodsList extends React.Component {
+  state = {
+    isReversed: false,
+    sortBy: null,
+    length: '1',
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: null,
+      length: '1',
+    });
+  }
+
+  sortByAlphabet = () => {
+    this.setState({ sortBy: 'alphabet' });
+  }
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  }
+
+  changeLength = (event) => {
+    this.setState({ length: event.target.value });
+  }
+
+  render() {
+    const { sortBy, isReversed, length } = this.state;
+    const resultList = [...this.props.list].filter(
+      word => word.length >= +length,
+    );
+
+    resultList.sort((a, b) => {
+      switch (sortBy) {
+        case 'alphabet':
+          return a.localeCompare(b);
+        case 'length':
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      resultList.reverse();
+    }
+
+    return (
+      <>
+        <div
+          style={{
+            display: 'flex', gap: '10px',
+          }}
+        >
+          <button
+            type="button"
+            onClick={this.reset}
+          >
+            Reset
+          </button>
+
+          <button
+            type="button"
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          <button
+            type="button"
+            onClick={this.sortByAlphabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <label htmlFor="length-select">Select min word length:</label>
+          <select
+            id="length-select"
+            value={length}
+            onChange={this.changeLength}
+          >
+            <option value="1">1</option>
+            <option value="2">2</option>
```

Comment:
You could create an array of numbers and map it, for not creating so many lines of code